src/catalog/src/durable/persist.rs

```diff
@@ -1431,19 +1431,20 @@ impl DurableCatalogState for PersistCatalogState {
         Ok(())
     }
 
-    // TODO(jkosh44) For most modifications we delegate to transactions to avoid duplicate code.
-    // This is slightly inefficient because we have to clone an entire snapshot when we usually
-    // only need one part of the snapshot. A Potential mitigation against these performance hits is
-    // to utilize `CoW`s in `Transaction`s to avoid cloning unnecessary state.
-
```

Comment:
`prune_storage_usage` is the only modification left, so the comment doesn't feel that necessary anymore.